dlt/common/json/__init__.py

```diff
@@ -21,8 +21,15 @@
 
 TPuaDecoders = List[Callable[[Any], Any]]
 
+Serializable = Union[str, Dict[Any, Any]]
 
-def custom_encode(obj: Any) -> Union[str, Dict[Any, Any]]:
+Encoder = Callable[[Any], Union[Serializable, None]]
+"""
+Custom user-defined encoder function type
+"""
+
+
+def custom_encode(_custom_encoder: Union[Encoder, None], obj: Any) -> Serializable:
```

Comment:
for regular code, using `partial` and injecting the encoder into signature would be ideal. but this is one of the most intensive called functions in `dlt` so what about the following
- we keep the signature intact
- we check a global variable `_custom_encoder ` in line 60
- this variable is stored in **this** module (but maybe set via _orjson or _simplejeson - please figure out)

Reply:
implemented in 659cc5c. One downside of the global var is that state can possibly leak, e.g. the custom encoder is not only set on the imported impl. I am unsure of how much of an issue that is, as most likely the `json` module won't be used as a utility, but only set up once before pipeline exectuon.